rules/Php81/Rector/New_/MyCLabsConstructorCallToEnumFromRector.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+namespace Rector\Php81\Rector\New_;
+
+use PhpParser\Node;
+use PhpParser\Node\Expr\New_;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Name;
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\ObjectType;
+use Rector\Enum\ObjectReference;
+use Rector\Rector\AbstractRector;
+use Rector\ValueObject\MethodName;
+use Rector\ValueObject\PhpVersionFeature;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\Php81\Rector\New_\MyCLabsConstructorCallToEnumFromRector\MyCLabsConstructorCallToEnumFromRectorTest
+ */
+final class MyCLabsConstructorCallToEnumFromRector extends AbstractRector implements MinPhpVersionInterface
+{
+    private const MY_C_LABS_CLASS = 'MyCLabs\Enum\Enum';
+
+    private const DEFAULT_ENUM_CONSTRUCTOR = 'from';
+
+    public function __construct(
+        private readonly ReflectionProvider $reflectionProvider,
+    ) {
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [New_::class];
+    }
+
+    public function refactor(Node $node): ?Node
+    {
+        if ($node instanceof New_) {
+            return $this->refactorConstructorCallToStaticFromCall($node);
+        }
+
+        return null;
+    }
```

Comment:
`Node` is always `New_` once defined in `getNodeTypes()`:

```suggestion
    /**
     * @param New_ $node
     */
    public function refactor(Node $node): ?Node
    {
        return $this->refactorConstructorCallToStaticFromCall($node);
    }
```

Reply:
Yeah, I was aware of this scenario. I wasn't sure if the PHP overriding annotation was acceptable in this community. Adjustment applied.